conda-store-server/conda_store_server/server/auth.py

```diff
@@ -271,7 +275,23 @@ def get_entity_bindings(self, entity: schema.AuthenticationToken):
     def convert_roles_to_permissions(self, roles):
         permissions = set()
         for role in roles:
-            permissions = permissions | self.role_mappings[role]
+            # 'editor' is the new alias of 'developer'. The new name is
+            # preferred in user-visible settings (like 'role_mappings') and when
+            # calling the role-mappings HTTP APIs, but it's ALWAYS mapped to
+            # 'developer' in the database for compatibility reasons.
+            # Additionally, this code allows for legacy 'role_mappings' that
+            # used to specify the role as 'developer'. Because it's a
+            # user-visible setting, we cannot break compatibility here
+            assert role != "editor"  # must NEVER be 'editor' in the DB
```

Comment:
An assert outside of a test strikes me as odd. Looking into it, it seems like some compilation flags remove assert statements. If this check is valuable, I think raising an exception here would be more valuable. However, I am not sure that this represents an unrecoverable error, since we would still know what to do with that data. We could just handle it and log an warning that data that shouldn't be in the db is. Or if it really does represent an unrecoverable error, than raising an exception seems better than an assert.

Reply:
@dcmcand Changed to `raise`. It's not unrecoverable, but it's a mistake that's easy to make when potentially refactoring or adding new DB APIs.

For example, see the change in `conda-store-server/conda_store_server/api.py` in this PR. The old code would bypass `validate_role` because `.update` is a lowel-level API, which just skips ORM validation logic. That was caught by the tests, which is how I noticed. But we might not have good coverage for some new yet to be added APIs, so I want to make sure this is never a problem.